src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -417,6 +418,152 @@ public async Task CreateOrUpdatePullRequestCommentAsync(string pullRequestUrl, s
             }
         }
 
+        /// <summary>
+        ///     Returns the ID used to identify the maestro merge policies checks in a PR
+        /// </summary>
+        /// <param name="mergePolicyName">Name of the merge policy</param>
+        /// <param name="sha">Sha of the latest commit in the PR</param>
+        private string CheckRunId(MergePolicyEvaluationResult result, string sha)
+        {
+            return $"{MergePolicyConstants.MaestroMergePolicyCheckRunPrefix}{result.MergePolicyInfo.Name}-{sha}";
+        }
+
+        public async Task CreateOrUpdatePullRequestMergeStatusInfoAsync(string pullRequestUrl, IReadOnlyList<MergePolicyEvaluationResult> evaluations)
+        {
+            (string owner, string repo, int id) = ParsePullRequestUri(pullRequestUrl);
+            // Get the sha of the latest commit for the current PR
+            string prSha = (await Client.PullRequest.Get(owner, repo, id))?.Head?.Sha;
+            if (prSha == null) 
+            {
+                throw new InvalidOperationException("We cannot find the sha of the pull request");
+            }
+
+            // Get a list of all the merge policies checks runs for the current PR
+            List <CheckRun> existingChecksRuns = 
+                (await Client.Check.Run.GetAllForReference(owner, repo, prSha))
+                .CheckRuns.Where(e => e.ExternalId.StartsWith(MergePolicyConstants.MaestroMergePolicyCheckRunPrefix)).ToList();
+
+            var toBeAdded = evaluations.Where(e => existingChecksRuns.All(c => c.ExternalId != CheckRunId(e, prSha)));
+            var toBeUpdated = existingChecksRuns.Where(c => evaluations.Any(e => c.ExternalId == CheckRunId(e, prSha)));
+            var toBeDeleted = existingChecksRuns.Where(c => evaluations.All(e => c.ExternalId != CheckRunId(e, prSha)));
```

Comment:
In what cases would we delete existing checks?

Reply:
We were not sure if it is even possible but I though about following use case:
1. Subscription merge policy is set to standard
2. Merge is blocked because of requested-changes policy
3. Subscription merge policy is **changed** to all-checks 
4. Now there are two checks (requested-changes and downgrade) which are not in play anymore

Is this scenario valid? Are changes in subs merge policy applied to running existing pull requests?

Reply:
All in all, I believe that this code has to be prepared for possibility of outdated existing checks as we can't rule this future possibility out.

Reply:
I believe changes in a subscription's merge policy would apply to the current subscription. This is almost certainly true for a batched subscription.

This is a good case to cover.